src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/CompletionResolveHandlerTest.cs

```diff
@@ -44,163 +53,142 @@ public CompletionResolveHandlerTest()
-            {
-                InsertText = "DateTime",
-            };
-            AssociateRequest(LanguageServerKind.CSharp, requestedCompletionItem, CompletionRequestContextCache, LoggerProvider);
-            var resolvedCompletionItem = new CompletionItem()
-            {
-                TextEdit = originalEdit,
-            };
-            var requestInvoker = CreateRequestInvoker((method, languageServerName, completionItem) => resolvedCompletionItem);
-            var handler = new CompletionResolveHandler(requestInvoker, DocumentManager, DocumentMappingProvider, FormattingOptionsProvider, CompletionRequestContextCache, LoggerProvider);
+        private ITextStructureNavigatorSelectorService TextStructureNavigatorSelectorService { get; }
 
-            // Act
-            var result = await handler.HandleRequestAsync(requestedCompletionItem, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+        private TestDocumentManager DocumentManager { get; } = new();
 
-            // Assert
-            Assert.Same(originalEdit, result.TextEdit);
-        }
+        private TestLSPDocumentMappingProvider DocumentMappingProvider { get; } = new();
 
-        [Fact]
-        public async Task HandleRequestAsync_NonNullOriginalTextEdit_DoesNotRemapTextEdit()
-        {
-            // Arrange
-            var originalEdit = new TextEdit() { NewText = "original" };
-            var mappedEdit = new TextEdit() { NewText = "mapped" };
-            DocumentMappingProvider.AddMapping(originalEdit, mappedEdit);
+        private TestFormattingOptionsProvider FormattingOptionsProvider { get; } = new();
 
-            var requestedCompletionItem = new CompletionItem()
-            {
-                TextEdit = originalEdit,
-            };
-            AssociateRequest(LanguageServerKind.CSharp, requestedCompletionItem, CompletionRequestContextCache, LoggerProvider);
-            var resolvedCompletionItem = new CompletionItem()
-            {
-                InsertText = "DateTime",
-                TextEdit = originalEdit,
-            };
-            var requestInvoker = CreateRequestInvoker((method, languageServerName, completionItem) => resolvedCompletionItem);
-            var handler = new CompletionResolveHandler(requestInvoker, DocumentManager, DocumentMappingProvider, FormattingOptionsProvider, CompletionRequestContextCache, LoggerProvider);
+        private CompletionRequestContextCache CompletionRequestContextCache { get; } = new();
 
-            // Act
-            var result = await handler.HandleRequestAsync(requestedCompletionItem, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+        private Uri HostDocumentUri { get; }
 
-            // Assert
-            Assert.Same(originalEdit, result.TextEdit);
-        }
+        private TestTextBuffer TextBuffer { get; }
 
-        [Fact]
-        public async Task HandleRequestAsync_ResolvedNullTextEdit_Noops()
```

Comment:
I renamed the test for this scenario to `HandleRequestAsync_CSharpProjection_DoNotRemapNullTextEdit`, it can be found further down the file